alica_engine/include/engine/BasicPlan.h

```diff
@@ -27,6 +30,7 @@ class BasicPlan : private RunnableObject
     virtual void onInit(){};
     virtual void run(void* msg){};
     virtual void onTerminate(){};
+    virtual const std::unordered_map<int64_t, std::unordered_set<int64_t>>& getApplicationEntrypointContext(std::vector<int64_t> entryPointIds);
```

Comment:
as suggested by @abhi1287, let's use the signature:
`virtual bool getApplicationEntrypointContext(std::unordered_map<int64_t, std::unordered_set<int64_t>& entryPointMap) override;` instead to avoid dynamic memory allocations on each tick of the engine